modules/common/log/default.nix

```diff
@@ -0,0 +1,55 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{lib, ...}:
+with lib; {
+  options.ghaf.logging.client.enable = lib.mkOption {
+    description = ''
+      Enable logging client service. Currently we have grafana alloy
+      running as client which will upload system journal logs to
+      grafana loki running in admin-vm
+    '';
+    type = lib.types.bool;
+    default = false;
+  };
+
+  options.ghaf.logging.server.enable = lib.mkOption {
+    description = ''
+      Enable logging server option will add functionality in
+      device to upload logs to remote server.
+    '';
+    type = lib.types.bool;
+    default = false;
+  };
+
+  options.ghaf.logging.port = lib.mkOption {
+    description = ''
+      Assign port value to grafana alloy process service running in
+      admin-vm. Using this port value firewall allowed TCP port
+      policy will be created Ghaf internally to grafana loki
+      (by default in admin-vm). No external network firewall policies
+      are created
+    '';
+    type = lib.types.int;
+    default = 3100;
+  };
+
+  options.ghaf.logging.listener.port = lib.mkOption {
+    description = ''
+      Assign port value to grafana alloy source service running in
+      different clients. Using this port value firewall allowed TCP port
```

Comment:
The grammar is a bit broken here. Maybe just "opens that port in the firewall"?

Though technically I disagree with doing this, I'd feel more comfortable if this would be done somewhere closer in the machine configs, and only for the network interfaces we need to accept traffic to.

Reply:
Can you please suggest where it can done?

Reply:
Instead of opening the firewall in this module, I'd expect the part setting `ghaf.logging.listener.port` in the machine config to also contain a statement opening the firewall.

Reply:
Done.